test/location.test.ts

```diff
@@ -33,4 +39,81 @@ describe('browser/location', () => {
       })
     );
   });
+
+  it('blocking transitions should work', async () => {
+    await doUpdateLocation({
+      pathname: '/',
+      search: {}
+    }).run();
+
+    let unblock = requestConfirmationToUpdateLocation();
+    await doUpdateLocation({
+      pathname: '/foo',
+      search: { bar: 'baz' }
+    }).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/',
+        search: {}
+      })
+    );
+    await doResolvePendingUpdateLocation(false).run();
+    expect(await location.run().run()).toEqual(
+      right({
+        pathname: '/',
+        search: {}
+      })
+    );
+    unblock();
+
+    unblock = requestConfirmationToUpdateLocation();
+    await doUpdateLocation({
+      pathname: '/foo',
+      search: { bar: 'baz' }
+    }).run();
+    expect(await location.run().run()).toEqual(
```

Comment:
this was already tested a few lines up isn't it?

Reply:
I want to test that it isn't yet changed here since we are blocking transitions. Technically yes, the same condition is tested a few lines above too, but this makes the test more readable IMO?